modules/renter/repair.go

```diff
@@ -195,10 +245,12 @@ func (f *file) offlineChunks(hdb hostDB) map[uint64][]uint64 {
 	// mark all pieces belonging to offline hosts.
 	offline := make(map[uint64][]uint64)
 	for _, fc := range f.contracts {
-		if hdb.IsOffline(fc.IP) {
-			for _, p := range fc.Pieces {
-				offline[p.Chunk] = append(offline[p.Chunk], p.Piece)
-			}
+		host, ok := hdb.Host(fc.IP)
+		if !ok || !isOffline(host) {
+			continue
+		}
```

Comment:
I'm not sure how big of an implementation change it would be, but it should be asking the contractor here instead of the hostdb.